GVRf/Extensions/gvrf-physics/src/main/jni/engine/bullet/bullet_rigidbody.cpp

```diff
@@ -111,13 +168,31 @@ void BulletRigidBody::setCenterOfMass(const Transform *t) {
 }
 
 void BulletRigidBody::getWorldTransform(btTransform &centerOfMassWorldTrans) const {
-    centerOfMassWorldTrans = convertTransform2btTransform(owner_object()->transform())
+    Transform* trans = owner_object()->transform();
+
+    centerOfMassWorldTrans = convertTransform2btTransform(trans)
                              * m_centerOfMassOffset.inverse();
 }
 
 void BulletRigidBody::setWorldTransform(const btTransform &centerOfMassWorldTrans) {
-    convertBtTransform2Transform(centerOfMassWorldTrans * m_centerOfMassOffset,
-                                 owner_object()->transform());
+    Transform* trans = owner_object()->transform();
+    btTransform aux; getWorldTransform(aux);
+
+    if(std::abs(aux.getOrigin().getX() - prevPos.getOrigin().getX()) >= 0.1f ||
+       std::abs(aux.getOrigin().getY() - prevPos.getOrigin().getY()) >= 0.1f ||
+       std::abs(aux.getOrigin().getZ() - prevPos.getOrigin().getZ()) >= 0.1f)
+    {
+        mRigidBody->setWorldTransform(aux);
+        prevPos = aux;
+        //TODO: incomplete solution
```

Comment:
This part is about updating rigid bodies whose scene object transform were changed outside the physics engine, currently we trade one physics step for the updating and static object are not updated as the physics ignores them. The comment is about looking for a more elegant solution. It will be deleted on the next commit.